src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -84,6 +84,20 @@ protected ParentRunner(Class<?> testClass) throws InitializationError {
         validate();
     }
 
+    /**
+     * Constructs a new {@code ParentRunner} that will run the {@code TestClass}.
+     * @since 4.13
+     */
+    protected ParentRunner(TestClass testClass) throws InitializationError {
+        this.testClass = testClass;
+        validate();
+    }
+
+    /**
+     * @deprecated Please use {@link #ParentRunner(org.junit.runners.model.TestClass)}.
+     * @since 4.12
+     */
+    @Deprecated
```

Comment:
I was going through open issues seeing what can/should be done in preparation of 4.13, and I noticed that this change wouldn't break people that extend `Theories`. It could break people that extend `BlockJUnit4ClassRunnerWithParameters and override `createTestClass()`.

I did some research. The `createTestClass()` method was so a custom runner could return a sub-class of `TestClass`  (see https://github.com/junit-team/junit4/pull/760#issuecomment-27836350 for the context).
(As an aside, calling non-static methods from the constructor should be avod

So for this change to break someone, all of the following would have to be true:

1. You would need to use `@ParametersRunnerFactory`
2. Your `ParametersRunnerFactory` implementation would have to create a subclass of `BlockJUnit4ClassRunnerWithParameters`
3. Your subclass of `BlockJUnit4ClassRunnerWithParameters` would have to override `createTestClass()` and return a custom subclass

Note that this could only happen in projects that use JUnit 4.12, as these APIs didn't exist before that.

It's unfortunate that we decided to create `createTestClass()`. The changes proposed in *this* pull would also allow a custom runner to use a custom sub-class of `TestClass`. Also, `createTestClass()` uses the anti-pattern of having a class override a method in the base class that is called in the parent constructor.

In any case, upon further reflection I'm leaning toward merging this pull, with one suggested change.